src/preamble.js

```diff
@@ -595,9 +595,12 @@ var wasmBinaryFile;
 if (Module['locateFile']) {
 #endif
   wasmBinaryFile = '{{{ WASM_BINARY_FILE }}}';
+#if !SINGLE_FILE
   if (!isDataURI(wasmBinaryFile)) {
     wasmBinaryFile = locateFile(wasmBinaryFile);
   }
+#endif
```

Comment:
This change looks separable.   Maybe we can land this on its own (it might be code size win separate to the encoding update)?

Reply:
Ok, #21587